### PR TITLE
[Metricbeat] Fix cloudwatch metricset missing tags collection

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to fix cloudwatch metricset from missing tags when `tags.resource_type_filter` is given. When user tries to collect daily storage metrics and s3 bucket tags with the cloudwatch metricset, it will fail to collect tags. This is because for CloudWatch metrics from S3 buckets, they always have `StorageType` as dimension. Dimension and S3 bucket name are combined to be the `identifierValue` in cloudwatch metricset code. But when collecting tags, dimension `StorageType` is not collected. That's why there is a mismatch between `identifierValue` and `resourceTagMap`.

## Why is it important?

This PR will fix tags collection for all namespaces that always have a dimension reported in CloudWatch metrics.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally
1. Make sure the AWS account you use for testing has S3 buckets with tags.
2. Use config below for aws.yml:
```
- module: aws
  period: 86400s
  metricsets:
    - cloudwatch
  metrics:
    - namespace: AWS/S3
      tags.resource_type_filter: s3
```
3. After starting Metricbeat, you should see tags collected from S3 buckets.

## Related issues

- Closes https://github.com/elastic/beats/issues/17419

### CODE REVIEW

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch_test.go [hunk 1]
```diff
@@ -1306,3 +1309,59 @@ func TestCreateEventsWithoutIdentifier(t *testing.T) {
...
+	}
+
+	cases := []struct {
+		title            string
+		identifier       string
+		expectedTagKey   string
+		expectedTagValue string
+	}{
+		{
+			"test identifier with only s3 bucket name",
```
Comment (ChrsMark) [inline x-pack/metricbeat/module/aws/cloudwatch/cloudwatch_test.go:1346]: Shouldn't this use `identifier2`?

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go [hunk 2]
```diff
@@ -567,3 +561,21 @@ func compareAWSDimensions(dim1 []cloudwatch.Dimension, dim2 []cloudwatch.Dimensi
...
+func insertTags(events map[string]mb.Event, identifier string, resourceTagMap map[string][]resourcegroupstaggingapi.Tag) {
+	// Check if identifier includes dimensionSeparator (comma in this case),
+	// split the identifier and check for each sub-identifier.
+	// For example, identifier might be [storageType, s3BucketName].
+	// And tags are only store under s3BucketName in resourceTagMap.
+	subIdentifiers := strings.Split(identifier, dimensionSeparator)
+	for _, v := range subIdentifiers {
+		tags := resourceTagMap[v]
+		if len(tags) != 0 {
+			// By default, replace dot "." using under bar "_" for tag keys and values
```
Comment (ChrsMark) [inline x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go:574]:
```suggestion
			// By default, replace dot "." using underscore "_" for tag keys and values
```
:)

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch_test.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/cloudwatch/cloudwatch_test.go:1346]: hmmm sorry I probably didn't get what you mean here. `identifier2` is in the next test case in Ln 1353.

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go (hunk 2 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go]: done! thanks 😄

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch_test.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/cloudwatch/cloudwatch_test.go]: sorry I finally realized whats wrong here. The title of the two test cases are flipped... Thanks for catching this. Fixed now!

**Pull request merged**